src/Cafe/HW/Latte/Renderer/Vulkan/VKRMemoryManager.h

```diff
@@ -138,7 +139,10 @@ class VKRMemoryManager
 {
 	friend class VKRSynchronizedRingAllocator;
 public:
-	VKRMemoryManager(class VulkanRenderer* renderer) : m_stagingBuffer(renderer, this, VKRSynchronizedRingAllocator::BUFFER_TYPE::STAGING, 32u * 1024 * 1024), m_indexBuffer(renderer, this, VKRSynchronizedRingAllocator::BUFFER_TYPE::INDEX, 4u * 1024 * 1024)
+	VKRMemoryManager(class VulkanRenderer* renderer) :
+			m_stagingBuffer(renderer, this, VKRSynchronizedRingAllocator::BUFFER_TYPE::STAGING, 32u * 1024 * 1024),
+			m_indexBuffer(renderer, this, VKRSynchronizedRingAllocator::BUFFER_TYPE::INDEX, 4u * 1024 * 1024),
+			m_vertexStrideMetalBuffer(renderer, this, VKRSynchronizedRingAllocator::BUFFER_TYPE::STRIDE, 4u * 1024 * 1024)
```

Comment:
Since this workaround is only required for MacOS, we should not allocate the stride buffer on other platforms.

Reply:
From reading the code of VKRSynchronizedRingAllocator it allocates the buffer on the first call to AllocateBufferMemory. And since that function is only ever called on macOS for m_vertexStrideMetalBuffer no additional memory should be allocated on different platforms. Is that okay or would you like to see some additional protections against accidental allocation?

Reply:
Mhh, yeah I forgot it does allocate on demand. It would be nice if we found a way to make it more explicit/obvious that nothing is being allocated. Maybe setting size to zero and then throwing an assert if an allocation is attempted?